esptool.py

```diff
@@ -653,7 +668,8 @@ def set_data_lengths(mosi_bits, miso_bits):
         if data_bits == 0:
             self.write_reg(SPI_W0_REG, 0)  # clear data register before we read it
         else:
-            data = pad_to(data, 4, b'\00')  # pad to 32-bit multiple
```

Comment:
The commit seems to have accidentally reverted a bunch of recent unrelated changes on master. Suggest doing something like a `git reset HEAD^`, then `git add -p` and stage only the relevant changes, then redo the commit as `git commit --reuse-message=2b4bd` and force push.